### PR TITLE
Fix unstable selected_rows_functor_test.cu

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: paddle/fluid/operators/math/selected_rows_functor.cu [hunk 1]
```diff
@@ -107,7 +107,7 @@ struct SelectedRowsAddTensor<platform::CUDADeviceContext, T> {
     PADDLE_ENFORCE_EQ(in1_height, out_dims[0]);
 
     auto& in1_value = input1.value();
-    framework::Vector<int64_t> in1_rows(input1.rows());
```
Comment (reyoung) [inline paddle/fluid/operators/math/selected_rows_functor.cu:110]: Prevent unnecessary memcpy here.

Not related to this bug.

File: paddle/fluid/operators/math/selected_rows_functor.cu [hunk 2]
```diff
@@ -206,7 +206,7 @@ struct SelectedRowsAddToTensor<platform::CUDADeviceContext, T> {
     PADDLE_ENFORCE_EQ(in1_height, in2_dims[0]);
 
     auto& in1_value = input1.value();
-    framework::Vector<int64_t> in1_rows(input1.rows());
```
Comment (reyoung) [inline paddle/fluid/operators/math/selected_rows_functor.cu:209]: Prevent unnecessary memcpy here.

Not related to this bug.

File: paddle/fluid/operators/math/selected_rows_functor_test.cu [hunk 3]
```diff
@@ -20,7 +20,9 @@ limitations under the License. */
 TEST(selected_rows_functor, gpu_add) {
   paddle::platform::CUDAPlace gpu_place(0);
   paddle::platform::CPUPlace cpu_place;
-  paddle::platform::CUDADeviceContext ctx(gpu_place);
+  paddle::platform::CUDADeviceContext& ctx =
+      *reinterpret_cast<paddle::platform::CUDADeviceContext*>(
+          paddle::platform::DeviceContextPool::Instance().Get(gpu_place));
```
Comment (reyoung) [inline paddle/fluid/operators/math/selected_rows_functor_test.cu:25]: Use the computation stream.

Comment (chengduoZH) [reply paddle/fluid/operators/math/selected_rows_functor_test.cu:25]: Why it must use the DeviceContextPool here?

**Review by chengduoZH — APPROVED** [review]
Cool

File: paddle/fluid/operators/math/selected_rows_functor_test.cu (hunk 3 above)
Comment (chengduoZH) [reply paddle/fluid/operators/math/selected_rows_functor_test.cu:25]: I see. The copy of lod is async and it is in computation stream.

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**